### PR TITLE
Adding CLI extension for Private DNS Public Preview

### PR DESCRIPTION
**Pull request by muwaqar**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [ ] Have you run `python scripts/ci/test_integration.py -q` locally?

### CODE REVIEW

**Comment by muwaqar** [comment]
@derekbekoe: `scripts/ci/test_integration.py` does not exist.
Also, are you the owner of `prodsbclipackages.blob.core.windows.net` or `azurecliprod.blob.core.windows.net`? We would like to host this extension wheel package there.

**Comment by muwaqar** [comment]
@derekbekoe @tjprescott : I have re-recorded the tests and they are passing now. The test that is failing is unrelated to my changes. Please have a look at the PR and if you do not have any comments, then please merge it at your earliest convenience.

File: src/dns/azext_dns/_params.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        c.argument('zone_name', options_list=('--zone-name', '-z'), help='The name of the zone.', type=dns_zone_name_type)
+        c.argument('metadata', nargs='+', help='Metadata in space-separated key=value pairs. This overwrites any existing metadata.', validator=validate_metadata)
+
+    with self.argument_context('network dns zone') as c:
+        c.argument('zone_name', name_arg_type)
+        c.ignore('location')
+
+        c.argument('zone_type', help='Type of DNS zone to create.', arg_type=get_enum_type(ZoneType))
+        c.argument('registration_vnets', arg_group='Private Zone', nargs='+', help='Space-separated names or IDs of virtual networks that register hostnames in this DNS zone.', validator=get_vnet_validator('registration_vnets'))
+        c.argument('resolution_vnets', arg_group='Private Zone', nargs='+', help='Space-separated names or IDs of virtual networks that resolve records in this DNS zone.', validator=get_vnet_validator('resolution_vnets'))
```
Comment (derekbekoe) [inline src/dns/azext_dns/_params.py]: There is no description for `--tags` in `az network dns zone create`

Comment (derekbekoe) [reply src/dns/azext_dns/_params.py]: Same for `az network dns zone update`

**Pull request merged**